### ISSUE TITLE
lookbehind doesn't process newlines or respect start position

### ISSUE THREAD

**Issue by burgerrg** [ds26gte/pregexp#2]
The regular expression `'.*(?<!/)'.*'` looks for strings with single quotes that have an internal single quote not preceded by a slash.

The following works as expected:
```
> (pregexp-match "'.*(?<!/)'.*'" "'/'fine'")
#f
```
Why does this happen?
```
> (pregexp-match "'.*(?<!/)'.*'" "\n'/'fine'")
("'/'fine'")
```

But it's fine with a non-newline:
```
> (pregexp-match "'.*(?<!/)'.*'" "\r'/'fine'")
#f
```

**Comment by burgerrg** [comment]
Here are tests that expose the bugs:
```
  (pregexp-match-positions "(?<=grey)hound"
    "\nthe hound in the picture is not a greyhound")
  ((39 . 44))

  (pregexp-match-positions "(?<!grey)hound"
    "\nthe greyhound in the picture is not a hound")
  ((39 . 44))

  (pregexp-match-positions "(?<=grey)hound"
    "greyhound"
    4)
  #f

  (pregexp-match-positions "(?<!grey)hound"
    "greyhound"
    4)
  ((4 . 9))
```

**Comment by burgerrg** [comment]
See pull request https://github.com/ds26gte/pregexp/pull/3

**Comment by ds26gte** [comment]
Thank you! I've pushed a rather smaller fix 43fd6da that should address both the issues that you point out.

**Comment by burgerrg** [comment]
Looks good, thank you!
